File: Core/SimpleDIContainer.vb
Public Class SimpleDIContainer
    Private ReadOnly container As New Dictionary(Of Type, Object)()

    Public Sub Register(serviceType As Type, implementation As Object)
        container(serviceType) = implementation
    End Sub

    Public Function Resolve(serviceType As Type) As Object
        If container.ContainsKey(serviceType) Then
            Return container(serviceType)
        End If
        ' Return Nothing if the service is not found.
        Return Nothing
    End Function

End Class

File: Core/SimpleDIContainer.cs
using System;
using System.Collections.Generic;

namespace AATM.Core
{
    public class SimpleDIContainer
    {
        private readonly Dictionary<Type, object> container = new Dictionary<Type, object>();

        public void Register(Type serviceType, object implementation)
        {
            container[serviceType] = implementation;
        }

        public object Resolve(Type serviceType)
        {
            if (container.ContainsKey(serviceType))
            {
                return container[serviceType];
            }
            // Return Nothing if the service is not found.
            return null;
        }

    }
}